docs/qldoc-style-guide.md

```diff
@@ -0,0 +1,184 @@
+# QLDoc style guide 
+
+## Introduction
+
+Valid QL comments are known as QLDoc. This document describes the recommended styles and conventions you should use when writing QLDoc for code contributions in this repository. If there is a conflict between any of the recommendations in this guide and clarity, then clarity should take precedence.
+
+### General requirements
+
+1. Documentation must adhere to the [QLDoc specification](https://help.semmle.com/QL/ql-handbook/qldoc.html).
+1. Documentation comments should be appropriate for users of the code.
+1. Documentation for maintainers of the code must use normal comments.
+1. Use `/** ... */` for documentation, even for single line comments.
+   - For single-line documentation, the `/**` and `*/` are written on the same line as the comment.
+   - For multi-line documentation, the `/**` and `*/` are written on separate lines. There is a `*` preceding each comment line, aligned on the first `*`.
+1. Use code formatting (`backticks`) within comments for code from the source language, and also for QL code (for example, names of classes, predicates, and variables).
+1. Give explanatory examples of code in the target language, enclosed in ```` ``` ````  or `` ` ``.
+
+
+### Language requirements
+
+1. Use US English.
+1. Use full sentences, with capital letters and full stops, except for the initial sentence of the comment, which may be fragmentary as described below.
```

Comment:
Isn't _full stop_ a British term? I think it's called a _period_ in U.S. English. Also, I think "US English" is written as "U.S. English" in U.S. English (with periods in the abbreviation).

(Sorry for being such a nit-picker. I don't know for sure whether I'm right here, but it would be slightly embarrassing if our requirement to use U.S. English isn't written in U.S. English :smile:.)

Reply:
👍  good spot on full stops vs periods.

And I think we should actually "American English" to avoid those troublesome abbreviations 😄